File: Leetcode/PY/Leetcode507-完美数.py
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        if num == 1: return False
        ans = 0
        for i in range(1,int(sqrt(num)) + 1):
            if num % i == 0: ans += i
            t = num // i
            if t != num and t != i and num % t == 0: ans += t
        return ans == num

File: Leetcode/Rust/Leetcode507-完美数.rs
impl Solution {
    pub fn check_perfect_number(num: i32) -> bool {
        let mut ans = 0;
        let mut i = 1;
        while i <= num / i {
            if num % i == 0{
                if i < num {
                    ans += i;
                }
                if num / i < num && i != num / i {
                    ans += num / i
                }
            }
            i += 1;
        }
        ans == num
    }
}